### PR TITLE
Update AE and VAE documentation

### PR DESCRIPTION
**Pull request by jeremyadamsfisher**
## What does this PR do?

Previously, pretrained models were initialized with a static method. Now, they are initialized with an instance method. Upate docstring accordingly.

Fixes https://github.com/PyTorchLightning/pytorch-lightning-bolts/issues/543

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?

On the master and on this branch, I have the following test failures:

```
FAILED tests/models/test_scripts.py::test_cli_run_cpc[ --data_dir /home/j/Documents/pytorch-lightning-bolts/datasets --max_epochs 1 --limit_train_batches 2 --limit_val_batches 2 --batch_size 2 --encoder resnet18]
FAILED tests/models/self_supervised/test_models.py::test_byol - ValueError: not enough values to unpack (expected 3, got 2)
FAILED tests/models/self_supervised/test_scripts.py::test_cli_run_self_supervised_cpc[ --data_dir /home/j/Documents/pytorch-lightning-bolts/datasets --max_epochs 1 --max_steps 3 --fast_dev_run 1 --batch_size 2 --encoder resnet18 --num_workers 0]
```

They seem unrelated to this issue

- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
🚀🚀🚀

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]
@jeremyadamsfisher Thank you for your contribution! LGTM!

**Comment by akihironitta** [comment]
@jeremyadamsfisher Would you mind fixing other places, too?
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/c5b3c04c97239019e2a782189917276a43587be8/pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py#L30

**Comment by jeremyadamsfisher** [comment]
> @jeremyadamsfisher Would you mind fixing other places, too?
> 
> https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/c5b3c04c97239019e2a782189917276a43587be8/pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py#L30

Done.

Other parts of the code don't have this issue because they don't have the API suggested in issue https://github.com/PyTorchLightning/pytorch-lightning-bolts/issues/200 . Here's where I've looked
- SimCLR (`SimCLR.load_from_checkpoint(weight_path, strict=False)`)
- CPCV2 (`CPCV2.load_from_checkpoint(weight_path, strict=False)` or `CPCV2(pretrained=True)`)
- SwAV (`SwAV.load_from_checkpoint(weight_path, strict=True)`)

**Review by akihironitta — APPROVED** [review]
@jeremyadamsfisher Thank you for your swift action!

File: pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ class AE(pl.LightningModule):
         ae = AE()
 
         # pretrained on cifar10
-        ae = AE.from_pretrained('cifar10-resnet18')
+        ae = AE(32).from_pretrained('cifar10-resnet18')
```
Comment (akihironitta) [inline pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py:30]:
```suggestion
        ae = AE(input_height=32).from_pretrained('cifar10-resnet18')
```

File: pl_bolts/models/autoencoders/basic_vae/basic_vae_module.py [hunk 2]
```diff
@@ -27,10 +27,10 @@ class VAE(pl.LightningModule):
         vae = VAE()
 
         # pretrained on cifar10
-        vae = VAE.from_pretrained('cifar10-resnet18')
+        vae = VAE(32).from_pretrained('cifar10-resnet18')
```
Comment (akihironitta) [inline pl_bolts/models/autoencoders/basic_vae/basic_vae_module.py:30]:
```suggestion
        vae = VAE(input_height=32).from_pretrained('cifar10-resnet18')
```

File: pl_bolts/models/autoencoders/basic_vae/basic_vae_module.py [hunk 3]
```diff
@@ -27,10 +27,10 @@ class VAE(pl.LightningModule):
...
 
         # pretrained on stl10
-        vae = VAE.from_pretrained('stl10-resnet18')
+        vae = VAE(32).from_pretrained('stl10-resnet18')
```
Comment (akihironitta) [inline pl_bolts/models/autoencoders/basic_vae/basic_vae_module.py:33]:
```suggestion
        vae = VAE(input_height=32).from_pretrained('stl10-resnet18')
```

**Review by Borda — APPROVED** [review]

**Pull request merged**